pkg/ndctl/ndctl.go

```diff
@@ -62,14 +66,18 @@ func (ctx *Context) GetBuses() []*Bus {
 
 //CreateNamespace create new namespace with given opts
 func (ctx *Context) CreateNamespace(opts CreateNamespaceOpts) (*Namespace, error) {
+	var err error
+	var ns *Namespace
 	for _, bus := range ctx.GetBuses() {
 		for _, r := range bus.ActiveRegions() {
-			if ns, err := r.CreateNamespace(opts); err == nil {
+			if ns, err = r.CreateNamespace(opts); err == nil {
```

Comment:
Here you get a fresh error, straight from `r.CreateNamespace`. Whatever was in err previously, like an error from a previous loop iteration, is lost here.